presto-hive/src/test/java/com/facebook/presto/hive/TestHiveLogicalPlanner.java

```diff
@@ -138,7 +142,117 @@ public void testPushdownFilter()
                 plan -> assertTableLayout(plan, "lineitem", withColumnDomains(ImmutableMap.of(new Subfield("partkey", ImmutableList.of()), Domain.singleValue(BIGINT, 10L))), remainingPredicate, ImmutableSet.of("partkey", "orderkey")));
     }
 
-    private RowExpression constant(int value)
+    @Test
+    public void testSubfieldExtraction()
+    {
+        Session pushdownFilterEnabled = Session.builder(getQueryRunner().getDefaultSession())
+                .setCatalogSessionProperty(HIVE_CATALOG, PUSHDOWN_FILTER_ENABLED, "true")
+                .build();
+        assertUpdate("DROP TABLE IF EXISTS test_subfield_extraction");
+        assertUpdate("CREATE TABLE test_subfield_extraction(" +
+                         "id bigint, " +
+                         "a array(bigint), " +
+                         "b map(varchar, bigint), " +
+                         "c row(" +
+                             "a bigint, " +
+                             "b row(x bigint), " +
+                             "c array(bigint), " +
+                             "d map(bigint, bigint), " +
+                             "e map(varchar, bigint)))");
+        PlanMatchPattern test_subfield_extaction_scan = anyTree(tableScan("test_subfield_extraction"));
+
+        assertPlan(pushdownFilterEnabled, "SELECT * FROM test_subfield_extraction where id = 2",
+                test_subfield_extaction_scan,
+                plan -> assertTableLayout(
+                        plan,
+                        "test_subfield_extraction",
+                        withColumnDomains(ImmutableMap.of(new Subfield("id", ImmutableList.of()), Domain.singleValue(BIGINT, 2L))),
```

Comment:
this can be simplified to `new Subfield("[id]")`

Reply:
"[id]" produces 

`com.facebook.presto.spi.PrestoException: Invalid subfield path: '[id‸]'

	at com.facebook.presto.spi.SubfieldTokenizer.invalidSubfieldPath(SubfieldTokenizer.java:262)
	at com.facebook.presto.spi.SubfieldTokenizer.matchUnquotedSubscript(SubfieldTokenizer.java:175)
	at com.facebook.presto.spi.SubfieldTokenizer.computeNext(SubfieldTokenizer.java:109)
	at com.facebook.presto.spi.SubfieldTokenizer.tryToComputeNext(SubfieldTokenizer.java:69)
	at com.facebook.presto.spi.SubfieldTokenizer.hasNext(SubfieldTokenizer.java:63)
	at com.facebook.presto.spi.Subfield.<init>(Subfield.java:217)
	at com.facebook.presto.hive.TestHiveLogicalPlanner.lambda$testSubfieldExtraction$7(TestHiveLogicalPlanner.java:167)
	at com.facebook.presto.tests.AbstractTestQueryFramework.lambda$assertPlan$6(AbstractTestQueryFramework.java:359)
	at com.facebook.presto.transaction.TransactionBuilder.execute(TransactionBuilder.java:150)
	at com.facebook.presto.tests.AbstractTestQueryFramework.assertPlan(AbstractTestQueryFramework.java:356)
	at com.facebook.presto.hive.TestHiveLogicalPlanner.testSubfieldExtraction(TestHiveLogicalPlanner.java:165)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:104)
	at org.testng.internal.Invoker.invokeMethod(Invoker.java:645)
	at org.testng.internal.Invoker.invokeTestMethod(Invoker.java:851)
	at org.testng.internal.Invoker.invokeTestMethods(Invoker.java:1177)
	at org.testng.internal.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:129)
	at org.testng.internal.TestMethodWorker.run(TestMethodWorker.java:112)
	at org.testng.TestRunner.privateRun(TestRunner.java:756)
	at org.testng.TestRunner.run(TestRunner.java:610)
	at org.testng.SuiteRunner.runTest(SuiteRunner.java:387)
	at org.testng.SuiteRunner.runSequentially(SuiteRunner.java:382)
	at org.testng.SuiteRunner.privateRun(SuiteRunner.java:340)
	at org.testng.SuiteRunner.run(SuiteRunner.java:289)
	at org.testng.SuiteRunnerWorker.runSuite(SuiteRunnerWorker.java:52)
	at org.testng.SuiteRunnerWorker.run(SuiteRunnerWorker.java:86)
	at org.testng.TestNG.runSuitesSequentially(TestNG.java:1293)
	at org.testng.TestNG.runSuitesLocally(TestNG.java:1218)
	at org.testng.TestNG.runSuites(TestNG.java:1133)
	at org.testng.TestNG.run(TestNG.java:1104)
	at org.testng.IDEARemoteTestNG.run(IDEARemoteTestNG.java:73)
	at org.testng.RemoteTestNGStarter.main(RemoteTestNGStarter.java:123)
`
"id" works though which intuitively makes sense to me.